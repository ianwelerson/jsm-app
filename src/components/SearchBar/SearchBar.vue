<script setup lang="ts">
import { watch } from 'vue'
import { useDebouncedRef } from '@/utils'
import IconBase from '../IconBase/IconBase.vue'

const searchTerm = useDebouncedRef('', 600)

watch(searchTerm, () => {
  triggerUpdate()
})

const triggerUpdate = () => {
  emit('update', searchTerm.value)
}

const emit = defineEmits<{
  (e: 'update', term: string): void
}>()
</script>

<template>
  <div class="search-bar">
    <div class="search-bar__content">
      <label class="search-bar__wrapper" for="search-bar">
        <div class="search-bar__icon">
          <IconBase name="IconSearch" :height="18" color="#323232" />
        </div>
        <input
          data-testid="search-bar-input"
          class="search-bar__input"
          id="search-bar"
          name="search-term"
          placeholder="Buscar aqui"
          type="text"
          v-model="searchTerm"
        />
      </label>
    </div>
  </div>
</template>

<style lang="scss" scoped>
.search-bar {
  width: 100%;

  &__wrapper {
    align-items: center;
    background-color: $white;
    border: $border-size-base solid $gray-600;
    border-radius: $radius-extra;
    display: flex;
    padding: $spacing-1 $spacing-5;
  }

  &__icon {
    margin-right: $spacing-2;
  }

  &__input {
    background-color: $white;
    border: none;
    outline: none;
    width: 100%;

    &::placeholder {
      color: $gray-700;
      font-size: $text-base;
      line-height: $leading-4;
    }
  }
}
</style>
