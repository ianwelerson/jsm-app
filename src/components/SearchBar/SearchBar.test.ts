import { render, fireEvent, waitFor } from '@testing-library/vue'
import SearchBar from '@/components/SearchBar/SearchBar.vue'

describe('SearchBar', () => {
  it('should emit an update with the field value when user type', async () => {
    const { getByTestId, emitted } = render(SearchBar)

    const searchInput = getByTestId('search-bar-input')

    await fireEvent.update(searchInput as HTMLInputElement, 'My term')

    expect((searchInput as HTMLInputElement).value).toEqual('My term')

    await waitFor(() => {
      expect(emitted('update')[0]).toEqual(['My term'])
    })
  })

  it('should match the snapshot', async () => {
    const { html } = render(SearchBar)

    expect(html()).toMatchSnapshot()
  })
})
