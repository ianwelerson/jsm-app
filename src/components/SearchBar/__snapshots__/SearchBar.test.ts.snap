// Vitest Snapshot v1

exports[`SearchBar > should match the snapshot 1`] = `
"<div class=\\"search-bar\\" data-v-e8d9439a=\\"\\">
  <div class=\\"search-bar__content\\" data-v-e8d9439a=\\"\\"><label class=\\"search-bar__wrapper\\" for=\\"search-bar\\" data-v-e8d9439a=\\"\\">
      <div class=\\"search-bar__icon\\" data-v-e8d9439a=\\"\\">
        <div data-testid=\\"icon-base\\" class=\\"icon-base\\" data-v-2336b232=\\"\\" data-v-e8d9439a=\\"\\">
          <!---->
        </div>
      </div><input data-testid=\\"search-bar-input\\" class=\\"search-bar__input\\" id=\\"search-bar\\" name=\\"search-term\\" placeholder=\\"Buscar aqui\\" type=\\"text\\" data-v-e8d9439a=\\"\\">
    </label></div>
</div>"
`;
